pydantic/decorators.py

```diff
@@ -344,22 +344,10 @@ def dec(f: Callable[..., Any] | classmethod[Any, Any, Any] | staticmethod[Any, A
     return dec
 
 
-_PlainSerializationFunction = Union[
-    _core_schema.GeneralPlainSerializerFunction,
-    _core_schema.FieldPlainSerializerFunction,
-    _decorators.GenericPlainSerializerFunctionWithoutInfo,
-    _decorators.FieldPlainSerializerFunctionWithoutInfo,
-    _PartialClsOrStaticMethod,
-]
+_PlainSerializationFunction = Union[_core_schema.SerializerFunction, _PartialClsOrStaticMethod]
 
 
-_WrapSerializationFunction = Union[
-    _core_schema.GeneralWrapSerializerFunction,
-    _core_schema.FieldWrapSerializerFunction,
-    _decorators.GeneralWrapSerializerFunctionWithoutInfo,
-    _decorators.FieldWrapSerializerFunctionWithoutInfo,
-    _PartialClsOrStaticMethod,
-]
+_WrapSerializationFunction = Union[_core_schema.WrapSerializerFunction, _PartialClsOrStaticMethod]
```

Comment:
Are we dropping all of this type checking for users as well? I think it's quite valuable.

Reply:
Is `_core_schema.WrapSerializerFunction` a union already? Sorry it's just hard to tell from the diff.